xbmc/video/VideoInfoScanner.cpp

```diff
@@ -1483,9 +1483,12 @@ namespace VIDEO
 
     if (CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
             CSettings::SETTING_MYVIDEOS_EXTRACTFLAGS) &&
-        CDVDFileInfo::GetFileStreamDetails(pItem))
+        !movieDetails.HasStreamDetails())
```

Comment:
I would swap the two terms, so that we evaluate from cheap to expensive (CPU cycles wise). `movieDetails.HasStreamDetails()` is definitely cheaper than the settings value retrieval.

=> `if (!movieDetails.HasStreamDetails() && CServicebroker::Get...`

Reply:
Fair.  Changed.